app/src/main/java/org/simple/clinic/appupdate/criticalupdatedialog/CriticalAppUpdateUiRenderer.kt

```diff
@@ -12,5 +14,14 @@ class CriticalAppUpdateUiRenderer(
     } else {
       ui.hideHelp()
     }
+
+    if (model.isCriticalUpdateNudgePriority && model.hasAppStaleness)
+      ui.renderCriticalAppUpdateReason(appStalenessInMonths(model.appStaleness!!))
+  }
+
+  private fun appStalenessInMonths(appStaleness: Int): Int {
+    val currentDate = LocalDate.now()
```

Comment:
Instead of creating the current date object here, should we instead pass it from the constructor? Because this method is now impure and we cannot set a specific value in our tests as well. WDYT? 

(_side note: We should also use `UserClock` when getting the current date_)

Reply:
We can also explore `Days` API from threeten extra library. We can use that to get the period and then convert it to the number of months. Like so

```kt
Days.of(appStaleness).toPeriod().months
```

With this, we don't have to construct a `LocalDate` object. You can take a look at whether this works for us.

Reply:
We cannot use this `Days.of(appStaleness).toPeriod().months` as the `period.months` method basically returns the months in the periods object and does not calculate the months from days. I have added `currentDate` as part of the constructor.

Reply:
Cool, sounds good 👍🏾

Reply:
I think we can also use `Months.from(Days.of(appStaleness)).amount`. But this works. Merging this in 👍🏾